### PR TITLE
Split X11 and Lomiri backends

### PR DESCRIPTION
**Pull request by tari01**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -39,7 +39,17 @@ add_definitions (-DGETTEXT_PACKAGE="${GETTEXT_PACKAGE}" -DLOCALEDIR="${CMAKE_INS
...
 include (FindPkgConfig)
-pkg_check_modules(SERVICE_DEPS REQUIRED glib-2.0>=2.36 gio-2.0>=2.36 x11>=1.6.7 libxklavier>=5.4 libayatana-common>=0.9.3)
+set(DEPS glib-2.0>=2.36 gio-2.0>=2.36 libayatana-common>=0.9.3)
+
+if(EXISTS "${CMAKE_INSTALL_FULL_LIBDIR}/maliit/plugins/liblomiri-keyboard-plugin.so")
+    set(HAS_LOMIRI_KEYBOARD ON)
+    add_definitions(-DHAS_LOMIRI_KEYBOARD)
+else()
+    set(DEPS ${DEPS} x11>=1.6.7 libxklavier>=5.4)
+endif()
```
Comment (sunweaver) [inline CMakeLists.txt:49]: This if-else-clause is wrong IMHO. We want to build the keyboard indicator with both backends, so this should be:

```
# build Lomiri backend if Maliit has the lomiri-keyboard-plugin
if(EXISTS "${CMAKE_INSTALL_FULL_LIBDIR}/maliit/plugins/liblomiri-keyboard-plugin.so")
￼    set(HAS_LOMIRI_KEYBOARD ON)
￼    add_definitions(-DHAS_LOMIRI_KEYBOARD)
endif()

# always build X11 backend
set(DEPS ${DEPS} x11>=1.6.7 libxklavier>=5.4)
```

Furthermore, I have the question if the Lomiri backend support can be built without Maliit's liblomiri-keyboard-plugin.so in place. If so, then the availability of Lomiri keyboard support should be a runtime condition, not a buildtime condition, IMHO.

**Review by sunweaver — CHANGES_REQUESTED** [review]
Please make the Lomiri and the X11 backend always available. There is no need IMHO for a buildtime condition. This will break the multi-desktop-env use case of ayatana-indicator-keyboard in distros where ayatana-indicator-keyboard could be used for e.g. MATE _and_ Lomiri on the same system.

File: CMakeLists.txt (hunk 1 above)
Comment (tari01) [reply CMakeLists.txt:49]: Errm... Your suggestion would force the X11 dependencies on Lomiri. I don't think they would be happy about that.

Comment (sunweaver) [reply CMakeLists.txt:49]: Good point. And, maybe they have those libs anyway, already? From the DEB maintainer's point of view, we need to make it possible to serve both purpose (Lomiri and X11) with the same build of the keyboard indicator.

The only thing, that comes up is loading the backends at runtime dynamically dependent on which backend is needed.

We could then package the backends (some .so files) in separate bin:pkgs where the X11 backend's bin:pkg has the libx* as dependencies whereas the Lomiri backend's bin:pkg does not (but pulls in maalit's lomiri keyboard plugin).

We must not force ourselves to provide two separate / different binary blobs of the same indicator. This opens a can of worms packaging wise long term.

File: src/service.c [hunk 2]
```diff
@@ -387,14 +394,51 @@ static void onDispose(GObject *pObject)
...
+        m_pLibHandle = NULL;
+    }
+
     G_OBJECT_CLASS(indicator_keyboard_service_parent_class)->dispose(pObject);
 }
 
 static void indicator_keyboard_service_init(IndicatorKeyboardService *self)
 {
+    gchar *sLib = "libayatana-keyboard-x11.so";
+    GFile *pFile = g_file_new_for_path("/usr/lib/maliit/plugins/liblomiri-keyboard-plugin.so");
```
Comment (sunweaver) [inline src/service.c:409]: You sure, that this file is always in this location?

Should we make this more configurable for package maintainers, e.g. via a ``-DMALIIT_PLUGINS_DIR=...``?

I'll merge this PR anyway, but this hard-coded path smells like an upcoming problem.

**Review by sunweaver — APPROVED** [review]
I don't like the hard-coded maliit plugins path in your PR. However, please hook with @dobey (Rodney) and discuss a way how this path can be provided by Maliit's pkg-config file (or some such mechanism), possibly. Thanks.

File: src/service.c (hunk 2 above)
Comment (tari01) [reply src/service.c:409]: You can hold the merge until I get some more info on this...

Comment (tari01) [reply src/service.c:409]: Damn, two seconds late!

**Pull request merged**